core/templates/components/common-layout-directives/common-elements/promo-bar.component.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a promo bar that appears at the top of the
+ * screen. The bar is configurable with a message and whether the promo is
+ * dismissible.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { PromoBarBackendApiService } from 'services/promo-bar-backend-api.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+@Component({
+  selector: 'oppia-promo-bar',
```

Comment:
Why is this named `oppia-promo-bar` and not just `promo-bar` is that some new convention? cc @srijanreddy98

Reply:
Yes. https://docs.google.com/document/d/1cXLNMmq2Bs3driED4B8JFqZZ_ZihuKD_8RkCfd-0W40/edit?usp=sharing

Reply:
Okay, thanks.